src/Sylius/Bundle/ResourceBundle/EventListener/CsrfListener.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Sylius\Bundle\ResourceBundle\EventListener;
+
+use Symfony\Component\Form\Extension\Csrf\CsrfProvider\SessionCsrfProvider;
+use Symfony\Component\HttpKernel\Event\FilterControllerEvent;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+
+/**
+ * @author Geoffrey Brier <geoffrey.brier@gmail.com>
+ */
+class CsrfListener
+{
+    /** @var \Symfony\Component\Form\Extension\Csrf\CsrfProvider\SessionCsrfProvider */
+    protected $csrfProvider;
+
+    /** @var string */
+    protected $intention;
+
+    /**
+     * @param SessionCsrfProvider $csrfProvider
+     * @param string              $intention
+     */
+    public function __construct(SessionCsrfProvider $csrfProvider, $intention)
+    {
+        $this->csrfProvider = $csrfProvider;
+        $this->intention = $intention;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function onKernelController(FilterControllerEvent $event)
+    {
+        $request = $event->getRequest();
+
+        if (!$request->isMethod('DELETE')) {
+            return;
+        }
+
+        if ($request->request->get('_csrf_token', '') !== $this->csrfProvider->generateCsrfToken($this->intention)) {
+            throw new AccessDeniedHttpException();
```

Comment:
Don't use `AccessDeniedHttpException`, you should use `Symfony\Component\Security\Core\Exception\AccessDeniedException` and Symfony will cover rest.